symengine/ruby/ext/symengine/symengine_macros.c

```diff
@@ -0,0 +1,51 @@
+#include "symengine_macros.h"
+#include "symengine.h"
+
+void sympify(VALUE operand2, basic_struct **cbasic_operand2) {
+    basic_struct *num_basic, *den_basic;
+    VALUE new_operand2, num, den;
+
+    switch(TYPE(operand2)) {
+        case T_FIXNUM:
+        case T_BIGNUM:
+            new_operand2 = cbasic_alloc(c_integer);
+            Data_Get_Struct(new_operand2, basic_struct, *cbasic_operand2);
+            GET_SYMINTFROMVAL(operand2, *cbasic_operand2);
+            break;
+
+        case T_RATIONAL:
+            num = rb_funcall(operand2, rb_intern("numerator"), 0, NULL);
+            den = rb_funcall(operand2, rb_intern("denominator"), 0, NULL);
+
+            num_basic = basic_new_heap();
+            den_basic = basic_new_heap();
+
+            GET_SYMINTFROMVAL(num, num_basic);
+            GET_SYMINTFROMVAL(den, den_basic);
+
+            new_operand2 = cbasic_alloc(c_rational);
```

Comment:
Also I hadn't used `basic` instead of `basic_struct *` is, initially when I started writing the wrappers, I was using `Data_Make_Struct`which itself allocated space, and hence using `basic` with it caused errors. But now since we are not using `Data_Make_Struct` anymore, `basic` can be used again.